lisa/nic.py

```diff
@@ -147,26 +175,29 @@ def __len__(self) -> int:
         return len(self.nics)
 
     def append(self, next_node: NicInfo) -> None:
-        self.nics[next_node.upper] = next_node
+        self.nics[next_node.name] = next_node
 
     def is_empty(self) -> bool:
         return len(self.nics) == 0
 
     def get_unpaired_devices(self) -> List[str]:
-        return [x.upper for x in self.nics.values() if not x.lower]
-
-    def get_upper_nics(self) -> List[str]:
-        return list(self.nics.keys())
+        return [x.name for x in self.nics.values() if not x.lower]
 
     def get_lower_nics(self) -> List[str]:
-        return [x.lower for x in self.nics.values() if x.lower]
+        return [x.name for x in self.nics.values() if x.upper]
+
+    def get_upper_nics_info(self) -> Dict[str, NicInfo]:
```

Comment:
It can return a list of NicInfo. And we can switch from `get_upper_nics` to this method step by step.